#! /bin/sh -e
## 02_gcc43_fix.dpatch by Masayuki Hatta <mhatta@debian.org>
##
## All lines beginning with \`## DP:' are a description of the patch.
## DP: Fix for GCC 4.3

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
        -patch) patch -f --no-backup-if-mismatch --dry-run -p1 < $0 && patch -f --no-backup-if-mismatch -p1 < $0
;;
        -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

diff -urN zipios++-0.1.5.9+cvs.2007.04.28.orig/src/appendzip.cpp zipios++-0.1.5.9+cvs.2007.04.28/src/appendzip.cpp
--- zipios++-0.1.5.9+cvs.2007.04.28.orig/src/appendzip.cpp	2007-04-28 02:32:31.000000000 +0900
+++ zipios++-0.1.5.9+cvs.2007.04.28/src/appendzip.cpp	2007-04-28 02:32:46.000000000 +0900
@@ -5,6 +5,8 @@
 #include "zipios++/ziphead.h"
 #include "zipios++/zipheadio.h"
 
+#include <cstdlib>
+
 using namespace std ;
 using namespace zipios ;
 

